pkg/datastore/projectstore.go

```diff
@@ -201,15 +200,17 @@ func (s *projectStore) UpdateProjectRBACConfig(ctx context.Context, id string, r
 
 // UpdateProjectRBACRoles updates project rbac roles.
 func (s *projectStore) UpdateProjectRBACRoles(ctx context.Context, id string, roles []*model.ProjectRBACRole) error {
-	return ErrUnimplemented
+	return s.update(ctx, id, func(p *model.Project) error {
+		p.RbacRoles = roles
+		p.RbacRoles = p.FilterRBACRoles(false)
+		return model.ValidateRBACRoles(p.RbacRoles)
+	})
 }
 
 // UpdateProjectUserGroups updates project user groups.
 func (s *projectStore) UpdateProjectUserGroups(ctx context.Context, id string, groups []*model.ProjectUserGroup) error {
-	return ErrUnimplemented
-}
-
-// MigrateFromProjectRBACConfig migrate from ProjectRBACConfig to ProjectUserGroup.
-func (s *projectStore) MigrateFromProjectRBACConfig(ctx context.Context, id string) error {
-	return ErrUnimplemented
+	return s.update(ctx, id, func(p *model.Project) error {
```

Comment:
The logic inside the update function will be atomic so all the below logic should be included in this function.
- moving old p.Rbac to the new user-groups
- set the p.Rbac = nil

Reply:
Maps of all user groups are returned on the client-side. And I think we can simply update here since the client sends all user groups to the request like this.

```
p.UserGroups, p.Rbac = groups, nil
```